src/main/java/cn/nukkit/block/BlockBed.java

```diff
@@ -167,6 +176,19 @@ public boolean onActivate(@Nonnull Item item, Player player) {
             return true;
         }
 
+        if (player != null && !player.isCreative()) {
+            AxisAlignedBB checkMonsterArea = new SimpleAxisAlignedBB(b.x - 8, b.y - 6.5, b.z - 8, b.x + 9, b.y + 5.5, b.z + 9)
+                    .addCoord(footPart.getXOffset(), 0, footPart.getZOffset());
+
+            for (Entity entity : this.level.getCollidingEntities(checkMonsterArea)) {
+                if (!entity.isClosed() && preventSleepMonsterId.contains(entity.getNetworkId())) {
+                    player.sendTranslation(TextFormat.GRAY + "%tile.bed.notSafe");
+                    return true;
+                }
+                // TODO: Check hostile mob, Piglin (check wearing golden armor)
```

Comment:
The idea of `Entity.isPreventingSleep(Player)` above could check if the player is wearing golden armor ^^

Reply:
I found weird behavior in Bedrock Edition, Piglin doesn't prevent sleep even it is angry. (Not wearing golden armor)